plugins/explain.py

```diff
@@ -1,103 +1,40 @@
-                  'GitLab, if you want to participate in the migration '
-                  'discussion, please add information [at our GitLab wiki page]'
-                  '(https://github.com/coala/coala/wiki/GitLab).',
-        'google': 'Hey. This message was triggered because someone was too '
-                  'lazy to type this *again*. Don\'t take it personally. '
-                  'Please.\n\nWe all got to learn this: *use google*. Or '
-                  'duckduckgo. Anything. The search engine that earned your '
-                  'trust. You got a build error? Search for the first red '
-                  'thing and google it. You got an exception? *Read the '
-                  'message.* Search it. *Think.*\n\nKeep this in mind: *You*'
-                  'are sitting in front of the problem, not us. You will have '
-                  'a much easier time solving it. That\'s why you should try '
-                  'doing it first.',
-        'promotion': 'To become part of the coala developers team, there '
-                     'are a few steps you need to complete. The newcomer '
-                     'process is as follows:\nYou will start as a newcomer, '
-                     'which is kind of a trial. If you complete the following '
-                     'tasks, you will become a developer at coala:\n\n- run '
-                     'coala on a project of yours\n- merge a difficulty/'
-                     'newcomer Pull Request\n- review at least a difficulty/'
-                     'newcomer Pull Request\n- merge a difficulty/low Pull '
-                     'Request\n- review at least a difficulty/low or higher '
-                     'Pull Request'
-    }
+    files = glob.glob('plugins/templates/explanations/*.jinja2.md')
+    KNOWN_KEYS = []
+    for fname in files:
+        KNOWN_KEYS.append(fname.replace(
+            'plugins/templates/explanations/', ''
+        ).replace('.jinja2.md', ''))
 
     ERROR_MSG = (
         'Sorry, I only know about these things:\n- ' +
-        '\n- '.join(MSGS.keys())
+        '\n- '.join(KNOWN_KEYS)
     )
 
     @re_botcmd(pattern=r'^explain\s+(\w+)(?:\s+to\s+@?([\w-]+))?$',
                re_cmd_name_help='explain <term>',
                flags=re.IGNORECASE)
     def explain(self, msg, match):
         """Explain various terms."""  # Ignore QuotesBear
-        return ('{}'.format('@{}: \n'.format(match.group(2))
-                            if match.group(2) else '') +
-                self.MSGS.get(
-                    match.group(1).lower(),
-                    self.ERROR_MSG
-                ).format(bot_prefix=self.bot_config.BOT_PREFIX))
+        response = ''
+        filename = 'explanations/{}.jinja2.md'.format(match.group(1).lower())
+        if match.group(1).lower() in self.KNOWN_KEYS:
+            if match.group(2):
+                response += '@{}: \n'.format(match.group(2))
+            response += tenv().get_template(filename).render()
```

Comment:
you can pass in `username` here, and it can be None, and use jinja's `if` syntax.

Bugger it, pass in `username` to them all, as we have another issue about personalising these messages, so if the username is available, that task becomes a jinja-markdown-only task, so newcomer level.

Reply:
Hmm, actually, we should have more clarity here, we need the "actor" username, and the "target" username.  They should have a consistent variable name through all of the templates.